worker/uniter/runner/context/contextfactory.go

```diff
@@ -214,6 +214,16 @@ func (f *contextFactory) HookContext(hookInfo hook.Info) (*HookContext, error) {
 			relation.cache.InvalidateMember(hookInfo.RemoteUnit)
 		}
 		hookName = fmt.Sprintf("%s-%s", relation.Name(), hookInfo.Kind)
+		// Update the private address on the context from the relation settings
```

Comment:
this seems wrong, since ctx.privateAddress might be used as the basis for setting the private-address relation settings